src/sys/node/worker.ts

```diff
@@ -1,10 +1,8 @@
-import '@stencil/core/compiler';
-
+import * as coreCompiler from '@stencil/core/compiler';
```

Comment:
the only code change I needed to make - this appears to be fine in both the esbuild-based build and the normal build

Reply:
Are there any ramifications from moving this to a compile-time reference from a runtime-based one? How can we test this out?

Reply:
tbh I'm not sure, I think the `prerender` task may use the node worker but I'm not certain -- I've basically confirmed that common functionality works as-expected (i.e. running a dev server, building, testing, etc)

Reply:
I'm not either 😆 Can you do me a favor and spend a little time digging into that? I think it'd be beneficial for us to build some of that knowledge while we're in here

Reply:
After doing a bunch of reading and documenting for #5034 I think that a basic way to test this is to just confirm that work which is done on the worker is, you know, done!

One example is here:

https://github.com/ionic-team/stencil/blob/6e89bcc7aa88dcc0b073390d8729bc76c22af974/src/compiler/optimize/optimize-module.ts#L101-L115

this `prepareModule` function called on `sys.worker` will, assuming you don't have `maxConcurrentWorkers` set to 1 or something, dispatch to a worker on another thread. If you have `minifyJs: true` in your Stencil config then the actual minification happens in another thread because of this - however, there is error recovery where if the worker doesn't work it will (I think!) output unminified JS instead. So I think a good check if that if you run a build then your minification settings are actually respected and the output actually changes if toggle `minifyJs`

Reply:
So more concretely, in a sample project you could

1. build and pack this branch and then install it
2. run a non-minified build (put `minifyJs: false` in `stencil.config.ts`)
3. check-in `dist/` with `git add -f dist/` to so you can see a diff later
4. run a minified build

and then just confirm that files that were unminified are now minified

Reply:
I've run this locally and it all seems to be working correctly